sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -512,6 +581,18 @@ String retrieveErrorDetails(Iterable<AppendRowsContext> failedContext) {
             .map(StackTraceElement::toString)
             .collect(Collectors.joining("\n"));
       }
+
+      void postFlush() {
+        // If we got a response indicating an updated schema, recreate the client.
+        if (updatedTableSchema != null
+            && this.appendClientInfo != null
+            && this.appendClientInfo.hasSchemaChanged(updatedTableSchema)) {
```

Comment:
Not sure if this is expensive to perform, theoretically it is not needed since when setting updatedTableSchema it is already checked. 

We also provided something here:
https://github.com/googleapis/java-bigquerystorage/blob/main/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java#L157 

If that field is not null, it means a new schema appeared. We do the comparison based on timestamp.

Reply:
removed this